backend/src/utils/prometheusUtils.ts

```diff
@@ -39,8 +49,11 @@ export const callPrometheus = async (
         });
         res.on('end', () => {
           try {
-            const parsedData: PrometheusResponse = JSON.parse(rawData);
+            const parsedData = JSON.parse(rawData);
             fastify.log.info('Successful response from Prometheus.');
+            if (parsedData.status === 'error') {
+              reject({ response: parsedData.error });
```

Comment:
Hmmm -- I think we need a code here. What scenario caused this to be a case you added?

Also -- `reject` doesn't stop the function. `return;` after it.

Reply:
This is when I use an invalid query language or forget to pass the required props like `start` or `end` when doing the range query. The Prometheus server will give me 400 bad request error, I think I can set the code to `400`